src/wazuh_db/wdb_parser.c

```diff
@@ -6117,3 +6120,40 @@ int wdb_parse_agents_clear_vuln_cve(wdb_t* wdb, char* output) {
     }
     return ret;
 }
+
+int wdb_parse_agents_update_status_vuln_cve(wdb_t* wdb, char* input, char* output) {
+    cJSON *data = NULL;
+    const char *error = NULL;
+    int ret = OS_INVALID;
+
+    data = cJSON_ParseWithOpts(input, &error, TRUE);
+
+    if (!data) {
+        mdebug1("Invalid vuln_cve JSON syntax when updating status value.");
+        mdebug2("JSON error near: %s", error);
+        snprintf(output, OS_MAXSTR + 1, "err Invalid JSON syntax, near '%.32s'", input);
+    }
+    else {
+        cJSON* old_status = cJSON_GetObjectItem(data, "old_status");
+        cJSON* new_status = cJSON_GetObjectItem(data, "new_status");
+        // Required fields
+        if (!cJSON_IsString(old_status) || !cJSON_IsString(new_status)) {
```

Comment:
Would it be necessary here to add an extra check, like `old_status->valuestring != NULL` ?

Reply:
In newer implementations of cJSON (the actual one) **cJSON_CreateString** necessarily creates a string when setting the json object as **cJSON_String**. So, checking with **cJSON_IsString** is enough to know that there is a string allocated.
This is relatively new and old versions of scan-build don't update it. This is why some parts of the Wazuh code have a double check of valuestring, but this is not required anymore.

Reply:
Thank you for the explanation !